code/obj/item/kendo.dm

```diff
@@ -0,0 +1,250 @@
+
+        if((attacker.a_intent == defender.a_intent) && !defender.hasStatus("disorient"))
+            playsound(defender, "sound/impact_sounds/kendo_parry_[pick(1,2,3)].ogg", 50, 1)
+            attacker.do_disorient(0,0,0,0,10,1)
+            return 1
+
+        else if(defender.hasStatus("blocking"))
+            playsound(defender, "sound/impact_sounds/kendo_block_[pick(1,2)].ogg", 50, 1)
+            if(attacker.equipped())
+                defender.do_disorient((attacker.equipped().stamina_cost*1.5),0,0,0,0,1,null)
+            return 2
+        return 0
+
+    proc/stat_reset()
+        if(force != 5)
+            force = 5
+        else
+            return
+        stamina_damage = 10
+        stamina_cost = 5
+        item_state = "shinai-light"
+        src.setItemSpecial(/datum/item_special/simple/kendo_light)
+        src.buildTooltipContent()
+
+    intent_switch_trigger(mob/user as mob)
+        if(guard != user.a_intent)
+            change_guard(user,user.a_intent)
+
+    attack(mob/living/carbon/human/defender as mob, mob/living/carbon/human/attacker as mob)
+        if(ishuman(defender))
+            if(defender.equipped() && istype(defender.equipped(),/obj/item/shinai))
+                var/obj/item/shinai/S = defender.equipped()
+                var/parry_block = S.parry_block_check(attacker,defender)
+                if((parry_block == 1) || (parry_block == 2))
+                    attacker.do_disorient((attacker.equipped().stamina_damage),0,0,0,0,1,null)
+                    return
+                else
+                    if((attacker.a_intent=="disarm") && prob(20))
+                        if(defender.equipped())
+                            var/obj/item/I = defender.equipped()
+                            defender.u_equip(I)
+                            I.set_loc(defender.loc)
+                            var/target_turf = get_offset_target_turf(I.loc,rand(5)-rand(5),rand(5)-rand(5))
+                            SPAWN_DBG(1 DECI SECOND)
+                                I.throw_at(target_turf,3,1)
+                                defender.show_text("<b>[attacker] knocks the [I] right out of your hands!</b>","red")
+                                attacker.show_text("<b>You knock the [I] right out of [defender]'s hands!</b>","green")
+        ..()
+
+    attack_hand(mob/user as mob)
+        if(src.loc != user)
+            change_guard(user,user.a_intent)
+        ..()
+
+    /*throw_begin(atom/target)
+        stat_reset()
+        ..()*/
+
+    dropped(mob/user as mob)
+        ..()
+        stat_reset()
+
+//==========
+//Shinai Bag
+//==========
+
+/obj/item/shinai_bag
+    name = "shinai bag"
+    desc = "竹刀袋 : a tube-like back for holding two shinai."
+    wear_image_icon = 'icons/mob/back.dmi'
+    icon_state = "shinaibag-closed"
+    item_state = "shinaibag-closed"
+    flags = ONBACK | FPRINT | TABLEPASS
+    w_class = 4.0
+    var/open = 0
+    var/shinai = 2
+
+    proc/update_spront(var/mob/user)
```

Comment:
What is a `spront`?

Reply:
its a shitty alternative name for a sprite XD

Reply:
i changed it in the next commit, so this will be resolved when that's pushed :heart: